src/Searchlight/LinqExecutor.cs

```diff
@@ -62,7 +62,16 @@ public static IEnumerable<T> QueryCollection<T>(this SyntaxTree tree, IEnumerabl
                     break;
             }
 
-            return queryable;
+            // construct the return fetch result
+            var result = new FetchResult<T>
+            {
+                pageSize = tree.PageSize,
+                pageNumber = tree.PageNumber,
+                totalCount = queryable.Count(),
```

Comment:
This isn't correct.  This count number is the count prior to pagination.  So that means you'd have to execute the queryable prior to line 50.

Reply:
That's exactly right.  TotalCount should reflect the *total* number of items *prior to pagination*.